src/vcpkg/binarycaching.cpp

```diff
@@ -1360,6 +1367,46 @@ namespace vcpkg
 
 namespace
 {
+    DECLARE_AND_REGISTER_MESSAGE(BcPathsMustBeAbsolute, (), "", "expected arguments: path arguments for binary config strings must be absolute");
```

Comment:
This really makes me wonder whether the regex based solutions are better; this does seems pretty awful for code review and maintenance being so far out of context.

Reply:
I am not going to implement regex based solutions. I think it would be a major mistake.